RHDHBUGS-1228: Argo CD update (#2245)
redhat-developer/red-hat-developers-documentation-rhdh: modules/shared/proc-enable-the-argo-cd-plugin.adoc

----
+
[NOTE]
====
Avoid using a trailing slash in the `url`, as it might cause unexpected behavior.
====

. Add the following annotation to the entity{apos}s `catalog-info.yaml` file to identify the Argo CD applications.

> 

Reviewer: List continuation cannot follow an empty line.

```suggestion
```

  argocd/app-selector: '${ARGOCD_LABEL_SELECTOR}'
----

. (Optional) Add the following annotation to the entity{apos}s `catalog-info.yaml` file to switch between Argo CD instances as shown in the following example:

> 

Reviewer: ```suggestion
```

====

.Procedure
* To enable the Argo CD plugin, set the `disabled` property to `false` in your `dynamic-plugins.yaml` file as follows:
+
[IMPORTANT]
====
Argo CD contains multiple frontend and backend options you can choose from for your Argo CD configuration:

* frontend:
** `community-plugins/argocd`
** `roadiehq/argocd`
* backend:
** `community-plugins/argocd-backend`

> ** `roadiehq/argocd-backend`

Reviewer 2: I would suggest using full package references:

```suggestion
* frontend:
** `@backstage-community/plugin-argocd`
** `@roadiehq/backstage-plugin-argo-cd`
* backend:
** `@backstage-community/plugin-argocd-backend`
** `@roadiehq/backstage-plugin-argo-cd-backend`
```

> The recommended combination is to use `community-plugins/argocd` together with `community-plugins/argocd-backend`.

Reviewer 2: ```suggestion
The recommended combination is to use `@backstage-community/plugin-argocd` together with `@backstage-community/plugin-argocd-backend`.
```

> However, you can also combine `community-plugins/argocd` or `roadiehq/argocd` together with `roadiehq/argocd-backend`.

Reviewer 2: ```suggestion
However, you can also combine `@backstage-community/plugin-argocd` or `@roadiehq/backstage-plugin-argo-cd` together with `@roadiehq/backstage-plugin-argo-cd-backend`.
```

> However, you can also combine `community-plugins/argocd` or `roadiehq/argocd` together with `roadiehq/argocd-backend`.

Reviewer 2: Hi @deerskindoll, I have talked with @christoph-jerolimov about this and it would be good if we don't encourage plugin mixing, although it is possible right now. The plugins will likely drift apart in the future.

```suggestion
```

Would you mind updating also backend and frontend options to a different structure, that lists packages per plugin (ArgoCD (Community / Red Hat) and Roadie ArgoCD instead of frontend vs backend)?

For Scaffolder Argo, we can add a note with something like: There is a separate `@roadiehq/scaffolder-backend-argocd` that provides custom Argo CD actions for Scaffolder templates, and it requires `@roadiehq/backstage-plugin-argo-cd-backend` to be enabled (installed).

Note, one can install at the same time roadie and community argocd backend plugins only starting from RHDH 1.10

Reviewer 2: Extensions name these plugins as: ArgoCD <-> Roadie Argo CD
You can also find them listed with names in these docs:
https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.9/html-single/dynamic_plugins_reference/index
https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.9/html-single/dynamic_plugins_reference/index#community-plugins-migration-table

Author: on it!

Author: @dzemanov I checked the links. **Argo CD** refers to roadie (tech preview plugins) and **ArgoCD Backend** refers to community (community plugins migration table). there's also **Roadie ArgoCD Backend** which is roadie and **Scaffolder Backend Argo CD** ` which is also roadie (plugins remaining bundled in 1.9).

The names aren't confusing at all.
(╯°□°）╯︵ ┻━┻

Author: @dzemanov re: scaffolder actions. I decided to make it a little less detailed so that I don't overload the user with info, since there is going to be a doc dedicated to scaffolder actions anyway. let me know if it's ok.

Reviewer 2: The inconsistent naming may be due to autogeneration – are these tables automatically populated (@nickboldt  might know)? Other entries in the tables contain different plugins under the same name as well, for example `catalog-backend-module-github` and `scaffolder-backend-module-github` are all under GitHub name. Then there is however Events Backend Module GitHub name 🧐
